src/shared/Core/GenericHostProvider.cs

```diff
@@ -125,6 +126,19 @@ public override async Task<ICredential> GenerateCredentialAsync(InputArguments i
             return await _basicAuth.GetCredentialsAsync(uri.AbsoluteUri, input.UserName);
         }
 
+        public override async Task<ICredential> GetCredentialAsync(InputArguments input)
+        {
+            var credential = await base.GetCredentialAsync(input);

```

Comment:
So depending on the host, the password is sometimes a JSON Web Token? I suggest to add this in a comment.

Reply:
Addressed in comments for `auth_token` and `refresh_token` checks.